syncd/scripts/syncd_init_common.sh

```diff
@@ -119,74 +119,175 @@ config_syncd_cisco_8000()
+    echo "# Start of $message" >> $to_file
+    while read line
+    do
+        line=$( echo $line | xargs )
+        if [ ! -z "$line" ];then
+            if [ "${line::1}" == '#' ];then
+                echo $line >> $to_file
+            elif [ "$line" == "[Overwrite Section]" ];then
+                override=true
+                echo "# $line" >> $to_file
+                echo "Merge properties with override $override"
+            else
+                sedline=${line%=*}
+                if grep -q $sedline $to_file ;then
+                   if $override ;then
+                      echo "Override the config $(grep $sedline $to_file) with $line in $to_file"
+                      prop=${line:0:`expr index $line =`}
+                      sed -i "/$prop/d" $to_file
+                      echo $line >> $to_file
+                   else
+                      grepline=$(grep $sedline $to_file)
+                      if [ "${grepline::1}" == '#' ];then
+                         echo $line >> $to_file
+                      else
+                         echo "Keep the config $(grep $sedline $to_file) in $to_file"
+                      fi
+                   fi
+                else
+                   echo $line >> $to_file
+                fi
+            fi
+        fi
+    done < $from_file
+    echo "# End of $message" >> $to_file
+    echo "Merged $from_file to $to_file"
+}
 
-    if [ -f $PLATFORM_DIR/common_config_support ];then
-
-      PLATFORM_COMMON_DIR=/usr/share/sonic/device/x86_64-broadcom_common
-
-      cp -f $HWSKU_DIR/*.config.bcm /tmp
-      cp -f /etc/sai.d/sai.profile /tmp
-      CONFIG_BCM=$(find /tmp -name '*.bcm')
-      PLT_CONFIG_BCM=$(find $HWSKU_DIR -name '*.bcm')
-      SAI_PROFILE=$(find /tmp -name 'sai.profile')
-      sed -i 's+/usr/share/sonic/hwsku+/tmp+g' $SAI_PROFILE
-
-      #Get first three characters of chip id
-      readline=$(grep '0x14e4' /proc/linux-kernel-bde)
-      chip_id=${readline#*0x14e4:0x}
-      chip_id=${chip_id::3}
-      COMMON_CONFIG_BCM=$(find $PLATFORM_COMMON_DIR/x86_64-broadcom_${chip_id} -name '*.bcm')
-
-      if [ -f $PLATFORM_COMMON_DIR/x86_64-broadcom_${chip_id}/*.bcm ]; then
-         for file in $CONFIG_BCM; do
-             echo "" >> $file
-             echo "# Start of chip common properties" >> $file
-             while read line
-             do
-               line=$( echo $line | xargs )
-               if [ ! -z "$line" ];then
-                   if [ "${line::1}" == '#' ];then
-                       echo $line >> $file
+function merge_config_yml_files()
+{
+    to_file=$1
+    from_file=$2
+    message=$3
+    override=false
+    merged_cnt=0
+    echo "" >> $to_file
+    echo "# Start of $message" >> $to_file
+    while read line
+    do
+        line=$( echo $line | xargs )
+        if [ ! -z "$line" ];then
+            if [ "${line::1}" == '#' ];then
+                echo "        $line" >> $to_file
+            elif [ "$line" == "[Overwrite Section]"];then
```

Comment:
@geans-pin Overwrite section should be after [Normal Section], if we move overwrite section before [Normal Section], the logic here will not work and will overwrite all properties, is it right? Do we have a comment to remind this?

Reply:
Yes, good catch. Let's have a comment in the script and HLD

Reply:
Had added the comment in HLD for reminding this

Reply:
Also add comment in script

Reply:
Zhaohui,

I've add this checker for Low Inheritance Precedence section. I think we can remove the
comment with the fix. 

Geans

            elif [ "$line" == "[Low Inheritance Precedence]" ];then
                echo "Checking Low Inheritance property..."
                check_override=false
            else